lib/bookingsync/engine.rb

```diff
@@ -17,6 +17,14 @@ class Engine < ::Rails::Engine
             env['omniauth.strategy'].options[:client_options].ssl = {
               verify: ENV['BOOKINGSYNC_VERIFY_SSL'] != 'false'
             }
+
+            if BookingSync::Engine.support_multi_applications?
+              credentials = BookingSync::Engine::CredentialsResolver.new(env["HTTP_HOST"]).call
+              if credentials.present?
+                env['omniauth.strategy'].options[:client_id] = credentials[:client_id]
```

Comment:
would be better if `BookingSync::Engine::CredentialsResolver` returned something nicer than a hash, this should be a method call of `client_id`, like `credentials.client_id.